### PR TITLE
Added DatabaseMetaData.getDatabaseCompatabilityLevel

### PR DESCRIPTION
**Pull request by dhait**
Implemented a new function to SQLServerDatabaseMetaData to check database compatibility level.

### CODE REVIEW

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java [hunk 1]
```diff
@@ -120,6 +120,16 @@ public void testDriverVersion() throws SQLException, IOException {
         }
     }
 
+    @Test
+    public void testDatabaseCompatibilityLevel() throws SQLException {
+        try (Connection conn = getConnection()) {
+            SQLServerDatabaseMetaData dbmData = (SQLServerDatabaseMetaData) conn.getMetaData();
+            int compatibilityLevel = dbmData.getDatabaseCompatibilityLevel();
+            assertTrue(compatibilityLevel > 0);
+            assertTrue(compatibilityLevel < dbmData.getDatabaseMajorVersion()*100);
```
Comment (lilgreenbird) [inline src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java:129]: I don't think we can guarantee this is always going to be true. Also I couldn't find any documentation on correlation between database version and compatibility level I think it's best not to assume this here

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 2]
```diff
@@ -2537,6 +2537,22 @@ public boolean locatorsUpdateCopy() throws SQLException {
         checkClosed();
         return true;
     }
+
+    /* -------------- MSSQL-JDBC Extension methods  ---------------- */
+
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:2542]: please add javadoc header (see other public api's for format)

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java [hunk 3]
```diff
@@ -2537,6 +2537,22 @@ public boolean locatorsUpdateCopy() throws SQLException {
...
+    public int getDatabaseCompatibilityLevel() throws SQLException {
+        checkClosed();
+        String database = connection.getCatalog();
+        try (SQLServerResultSet rs = getResultSetFromInternalQueries(null,
+                "select name, compatibility_level from sys.databases where name = '" + database + "'")) {
+            if (!rs.next()) {
+                return 0;
+            }
+            return rs.getInt("compatibility_level");
+        } catch (SQLServerException e) {
```
Comment (lilgreenbird) [inline src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:2552]: remove and just let the error throw

File: src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java (hunk 1 above)
Comment (dhait) [reply src/test/java/com/microsoft/sqlserver/jdbc/databasemetadata/DatabaseMetaDataTest.java:129]: Here's the link to the MS documentation:
[https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver15](https://docs.microsoft.com/en-us/sql/t-sql/statements/alter-database-transact-sql-compatibility-level?view=sql-server-ver15).
You are correct that in the case of Azure, it is possible that the compatibility level is greater than the database engine version, so I've deleted that check.  It should however always be greater than 0.

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java (hunk 3 above)
Comment (dhait) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:2552]: Fixed

File: src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java (hunk 2 above)
Comment (dhait) [reply src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDatabaseMetaData.java:2542]: Done.

**Comment by dhait** [comment]
The failure on Windows build was TVPAllTypesTest.testTVPStoredProcedureResultSet(TVPAllTypesTest.java:134)
This is unrelated to my code changes.
Is it possible to investigate and rerun the test build?

**Comment by ulvii** [comment]
Thanks for the contribution @dhait .

**Pull request merged**